go2ts.go

```diff
@@ -416,18 +451,28 @@ func (g *Go2TS) populateInterfaceDeclarationProperties(interfaceDeclaration *typ
 		property := typescript.PropertySignature{
 			Identifier: propertyName,
 			Type:       propertyType,
-			Optional:   recursivelyForceOptional || markedAsOptional,
+			Optional:   optionalFieldPolicy == recursivelyForceOptional || markedAsOptional,
 		}
 		interfaceDeclaration.Properties = append(interfaceDeclaration.Properties, property)
 	}
 }
 
-func (g *Go2TS) reflectTypeToTypeScriptType(reflectType reflect.Type, namespace string, wasExplicitlyAdded, ignoreNil bool) typescript.Type {
+// wasExplicitlyAdded indicates whether a Go type was explicitly added to a Go2TS instance via one
+// of the Add* methods, or implicitly, e.g. by discovering a user-defined type when inspecting the
+// field types of a Go struct type explicitly added by the user.
+type wasExplicitlyAdded int
```

Comment:
nit: wasExplicitlyAdded isn't a great name, how about `typeDiscovery`, which more closely aligns with the comment above.

Reply:
Much better, thanks!